plasmapy/particles/_factory.py

```diff
@@ -98,19 +100,29 @@ def _physical_particle_factory(
     if not args and not kwargs:
         raise TypeError("Particle information has not been provided.")
 
-    for particle_type in (Particle, CustomParticle, ParticleList):
-        with contextlib.suppress(TypeError, InvalidParticleError):
+    try:
+        return Particle(*args, **kwargs)
+    except _particle_creation_exceptions:
+        # If the first argument is a string but is not a valid particle,
+        # then don't try to create a CustomParticle or ParticleList.
+        if args and isinstance(args[0], str):
+            raise
+
+    for particle_type in (CustomParticle, ParticleList):
+        with contextlib.suppress(*_particle_creation_exceptions):
             return particle_type(*args, **kwargs)
```

Comment:
Hm...it's looking like I'll need to restructure this part again in #1867 to enable `@particle_input` to take values for `Z` that are real but not integers.